bindata/etcd/cluster-backup.sh

```diff
@@ -60,23 +35,27 @@ function source_required_dependency {
   source "${path}"
 }
 
-BACKUP_DIR="$1"
-DATESTRING=$(date "+%F_%H%M%S")
-BACKUP_TAR_FILE=${BACKUP_DIR}/static_kuberesources_${DATESTRING}.tar.gz
-SNAPSHOT_FILE="${BACKUP_DIR}/snapshot_${DATESTRING}.db"
-BACKUP_RESOURCE_LIST=("kube-apiserver-pod" "kube-controller-manager-pod" "kube-scheduler-pod" "etcd-pod")
 
-trap 'rm -f ${BACKUP_TAR_FILE} ${SNAPSHOT_FILE}' ERR
+trap 'podman rm --force cluster-backup-$$' ERR
 
 source_required_dependency /etc/kubernetes/static-pod-resources/etcd-certs/configmaps/etcd-scripts/etcd.env
-source_required_dependency /etc/kubernetes/static-pod-resources/etcd-certs/configmaps/etcd-scripts/etcd-common-tools
 
-# TODO handle properly
-if [ ! -f "$ETCDCTL_CACERT" ] && [ ! -d "${CONFIG_FILE_DIR}/static-pod-certs" ]; then
-  ln -s "${CONFIG_FILE_DIR}"/static-pod-resources/etcd-certs "${CONFIG_FILE_DIR}"/static-pod-certs
-fi
+podman run  \
+  --name cluster-backup-$$ \
```

Comment:
just curious on the naming here I see that your adding the process id. Is the reasoning to support more than one backup container at time. Lets say some automation run 500 instances of this by accident. If we force a single name then that failure case would be stopped.

I dont feel overly strongly here but I would like to understand your reasoning on a funcitonal level.

Reply:
Actually more I think about this I would need a strong reason for doing this.

Reply:
Usually if something is temporary (ephemeral and only applicable to the process in question), it is customary to name it with the process-id, so that it is indicative of its temporary nature. 

Also, for some reason if we have an automated backup job running in the background (using CronJob or something like that), and an admin wants to take a backup at the same time, then, each of these two processes should work fine and result in a successful backup.

Reply:
Ultimately, the questions comes down to "do we want to allow simultaneous backups?" If the answer is yes, then, we need a process specific name.

Reply:
> Ultimately, the questions comes down to "do we want to allow simultaneous backups?" If the answer is yes, then, we need a process specific name.

Does this sound like a good idea to you?

Reply:
what possible implications are there for doing this?

Reply:
I think kube will handle scheduling/naming containers if this is run in a container such as cronjob so this usage would be manual only.

Reply:
I removed the $$, so that simultaneous invocation would error out.